include/mbedtls/platform_util.h

```diff
@@ -145,6 +145,17 @@ MBEDTLS_DEPRECATED typedef int mbedtls_deprecated_numeric_constant_t;
 #define MBEDTLS_IGNORE_RETURN(result) ((void) !(result))
 #endif
 
+/* This can be used to pack structs for memory savings.
+ *
+ * The compiler may generate larger/slower code to perform accesses to unaligned
+ * struct contents on architectures which do not support unaligned accesses.
+ */
```

Comment:
If we add that, we must make a note that this must never be used in a public header, since the library and applications may be compiled with different compilers.